shdlc/sensirion_shdlc.h

```diff
@@ -99,6 +105,163 @@ int16_t sensirion_shdlc_xcv(uint8_t addr, uint8_t cmd, uint8_t tx_data_len,
                             struct sensirion_shdlc_rx_header* rx_header,
                             uint8_t* rx_data);
 
+/**
+ * sensirion_shdlc_add_uint8_t_to_frame() - Add a uint8_t to the frame at
+ *                                          offset.
+ *
+ * @param tx_frame Pointer to buffer in which the write frame will be prepared.
+ *                 Caller needs to make sure that there is enough space left in
+ *                 the buffer pointed to by the data member after offset in
+ *                 tx_frame, to write the data.
+ * @param data     uint8_t to be written into the frame.
+ *
+ */
+void sensirion_shdlc_add_uint8_t_to_frame(
+    struct sensirion_shdlc_buffer* tx_frame, uint8_t data);
+
+/**
+ * sensirion_shdlc_begin_frame() - Initialize buffer and add the first three
+ *                                 fixed-use data bytes to it.
+ *
+ * @param tx_frame    Pointer to struct sensirion_shdlc_buffer, all members will
+ *                    be initialized in this function.
+ * @param buffer      The pointer to a big enough buffer to write the whole
+ *                    tx_frame. Caller needs to initialize this buffer.
+ * @param command     Command to be written into the frame.
+ * @param address     Address to be written into the frame.
+ * @param data_length Number of bytes of data to be written into the frame after
+ *                    this method.
+ */
+void sensirion_shdlc_begin_frame(struct sensirion_shdlc_buffer* tx_frame,
+                                 uint8_t* buffer, uint8_t command,
+                                 uint8_t address, uint8_t data_length);
+
+/**
+ * sensirion_shdlc_add_uint32_t_to_frame() - Add a uint32_t to the frame at
+ *                                           offset.
+ *
+ * @param tx_frame Pointer to buffer in which the write frame will be prepared.
+ *                 Caller needs to make sure that there is enough space left in
+ *                 the buffer pointed to by the data member after offset in
+ *                 tx_frame, to write the data.
+ * @param data     uint32_t to be written into the frame.
+ *
+ */
+void sensirion_shdlc_add_uint32_t_to_frame(
+    struct sensirion_shdlc_buffer* tx_frame, uint32_t data);
+
+/**
+ * sensirion_shdlc_add_int32_t_to_frame() - Add a int32_t to the frame at
+ *                                         offset.
```

Comment:
```suggestion
 *                                          offset.
```
also the next 2 functions